pywhat/regex_identifier.py

```diff
@@ -4,23 +4,34 @@
 import re
 from typing import Optional
 
-from pywhat.distribution import Distribution
+from pywhat.filter import Distribution, Filter
 
 
 class RegexIdentifier:
     def __init__(self):
         self.distribution = Distribution()
 
-    def check(self, text, distribution: Optional[Distribution] = None):
-        if distribution is None:
-            distribution = self.distribution
+    def check(
+        self,
+        text,
+        dist: Optional[Distribution] = None,
+        *,
+        boundaryless: Optional[Filter] = None
+    ):
+        if dist is None:
+            dist = self.distribution
+        if boundaryless is None:
+            boundaryless = Filter({"Tags": []})
         matches = []
 
         for string in text:
-            for reg in distribution.get_regexes():
-                matched_regex = re.search(reg["Regex"], string, re.UNICODE)
+            for reg in dist.get_regexes():
+                regex = reg["Regex"]
+                if reg in boundaryless:
+                    regex = re.sub(r"(?<!\\)\^(?![^\[\]]*(?<!\\)\])", "", regex)
+                    regex = re.sub(r"(?<!\\)\$(?![^\[\]]*(?<!\\)\])", "", regex)
```

Comment:
Why not make a distribution with boundaryless regex? :) Would be faster over repeated calls via API :)